<!DOCTYPE html>
<!-- Generated by pkgdown: do not edit by hand --><html lang="en">
<head>
<meta http-equiv="Content-Type" content="text/html; charset=UTF-8">
<meta charset="utf-8">
<meta http-equiv="X-UA-Compatible" content="IE=edge">
<meta name="viewport" content="width=device-width, initial-scale=1.0">
<title>Introduction to coca • coca</title>
<!-- jquery --><script src="https://cdnjs.cloudflare.com/ajax/libs/jquery/3.4.1/jquery.min.js" integrity="sha256-CSXorXvZcTkaix6Yvo6HppcZGetbYMGWSFlBw8HfCJo=" crossorigin="anonymous"></script><!-- Bootstrap --><link href="https://cdnjs.cloudflare.com/ajax/libs/bootswatch/3.4.0/yeti/bootstrap.min.css" rel="stylesheet" crossorigin="anonymous">
<script src="https://cdnjs.cloudflare.com/ajax/libs/twitter-bootstrap/3.4.1/js/bootstrap.min.js" integrity="sha256-nuL8/2cJ5NDSSwnKD8VqreErSWHtnEP9E7AySL+1ev4=" crossorigin="anonymous"></script><!-- bootstrap-toc --><link rel="stylesheet" href="../bootstrap-toc.css">
<script src="../bootstrap-toc.js"></script><!-- Font Awesome icons --><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/all.min.css" integrity="sha256-mmgLkCYLUQbXn0B1SRqzHar6dCnv9oZFPEC1g1cwlkk=" crossorigin="anonymous">
<link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/v4-shims.min.css" integrity="sha256-wZjR52fzng1pJHwx4aV2AO3yyTOXrcDW7jBpJtTwVxw=" crossorigin="anonymous">
<!-- clipboard.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/clipboard.js/2.0.6/clipboard.min.js" integrity="sha256-inc5kl9MA1hkeYUt+EC3BhlIgyp/2jDIyBLS6k3UxPI=" crossorigin="anonymous"></script><!-- headroom.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/headroom.min.js" integrity="sha256-AsUX4SJE1+yuDu5+mAVzJbuYNPHj/WroHuZ8Ir/CkE0=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/jQuery.headroom.min.js" integrity="sha256-ZX/yNShbjqsohH1k95liqY9Gd8uOiE1S4vZc+9KQ1K4=" crossorigin="anonymous"></script><!-- pkgdown --><link href="../pkgdown.css" rel="stylesheet">
<script src="../pkgdown.js"></script><meta property="og:title" content="Introduction to coca">
<meta property="og:description" content="coca">
<!-- mathjax --><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/MathJax.js" integrity="sha256-nvJJv9wWKEm88qvoQl9ekL2J+k/RWIsaSScxxlsrv8k=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/config/TeX-AMS-MML_HTMLorMML.js" integrity="sha256-84DKXVJXs0/F8OTMzX4UR909+jtl4G7SPypPavF+GfA=" crossorigin="anonymous"></script><!--[if lt IE 9]>
<script src="https://oss.maxcdn.com/html5shiv/3.7.3/html5shiv.min.js"></script>
<script src="https://oss.maxcdn.com/respond/1.4.2/respond.min.js"></script>
<![endif]--><!-- Global site tag (gtag.js) - Google Analytics --><script async src="https://www.googletagmanager.com/gtag/js?id=UA-128138627-4"></script><script>
  window.dataLayer = window.dataLayer || [];
  function gtag(){dataLayer.push(arguments);}
  gtag('js', new Date());

  gtag('config', 'UA-128138627-4');
</script>
</head>
<body data-spy="scroll" data-target="#toc">
    <div class="container template-article">
      <header><div class="navbar navbar-default navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-target="#navbar" aria-expanded="false">
        <span class="sr-only">Toggle navigation</span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <span class="navbar-brand">
        <a class="navbar-link" href="../index.html">coca</a>
        <span class="version label label-default" data-toggle="tooltip" data-placement="bottom" title="Released version">1.1.0</span>
      </span>
    </div>

    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
<li>
  <a href="../articles/index.html">
    <span class="fa fa-play-circle"></span>
     
    Vignettes
  </a>
</li>
<li>
  <a href="../reference/index.html">
    <span class="fa fa-pencil"></span>
     
    Documentation
  </a>
</li>
<li>
  <a href="https://github.com/acabassi/klic">
    <span class="fa fa-github fa-lg"></span>
     
    GitHub
  </a>
</li>
      </ul>
<ul class="nav navbar-nav navbar-right">
<li>
  <a href="http://github.com/acabassi/coca/">
    <span class="fab fa fab fa-github fa-lg"></span>
     
  </a>
</li>
      </ul>
</div>
<!--/.nav-collapse -->
  </div>
<!--/.container -->
</div>
<!--/.navbar -->

      

      </header><div class="row">
  <div class="col-md-9 contents">
    <div class="page-header toc-ignore">
      <h1 data-toc-skip>Introduction to coca</h1>
                        <h4 class="author">Alessandra Cabassi</h4>
            
            <h4 class="date">2020-07-08</h4>
      
      <small class="dont-index">Source: <a href="http://github.com/acabassi/coca/blob/master/vignettes/coca-vignette.Rmd"><code>vignettes/coca-vignette.Rmd</code></a></small>
      <div class="hidden name"><code>coca-vignette.Rmd</code></div>

    </div>

    
    
<div id="introduction" class="section level1">
<h1 class="hasAnchor">
<a href="#introduction" class="anchor"></a>Introduction</h1>
<p>The R package <strong>coca</strong> contains the functions needed to use COCA (<strong>C</strong>luster-<strong>O</strong>f-<strong>C</strong>lusters <strong>A</strong>nalysis), an integrative clustering method that was first introduced in a breast cancer study by The Cancer Genome Atlas in 2012 and quickly became a popular tool in cancer studies (see e.g. Hoadley et al. 2014 and Aure et al. 2017). It is based on Consensus Clustering (Monti et al., 2013), an algorithm that was initially developed to assess the stability of clusters obtained with any clustering algorithm.</p>
<p>The main goal of COCA is to summarise clusterings found in different ’omics datasets, by identifying a “global” clustering across the datasets that is intended to be in good agreement with the clustering structures identified in each of the individual datasets. For further details about the algorithm, please see Cabassi and Kirk (2019).</p>
</div>
<div id="building-the-matrix-of-clusters" class="section level1">
<h1 class="hasAnchor">
<a href="#building-the-matrix-of-clusters" class="anchor"></a>Building the matrix of clusters</h1>
<p>The first step of COCA is the construction of the Matrix-Of-Clusters (MOC). This is a binary matrix of size <em>N x K</em>, where <em>K</em> is the sum of the number of clusters <em>Km</em> in every dataset <em>Xm</em>. Therefore, to each column <em>j</em> of this matrix corresponds a cluster <em>mk</em> in dataset <em>Xm</em>. The <em>(i,j)</em>-th entry of the matrix of clusters is equal to one if data point <em>i</em> belongs to cluster <em>mk</em> in dataset <em>Xm</em>, and is equal to zero otherwise.</p>
<p>The function that can be used to build a matrix of clusters starting from a list of heterogeneous datasets (referring to the same observations) is <code>buildMOC</code>. In the example below, we assume to have three datasets with the same number of clusters, five. The clustering structure in each of those datasets is found via k-means clustering.</p>
<div class="sourceCode" id="cb1"><html><body><pre class="r"><span class="co">### Load data</span>
<span class="no">data</span> <span class="kw">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/list.html">list</a></span>()
<span class="no">data</span><span class="kw">[[</span><span class="fl">1</span>]] <span class="kw">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/matrix.html">as.matrix</a></span>(<span class="fu"><a href="https://rdrr.io/r/utils/read.table.html">read.csv</a></span>(<span class="fu"><a href="https://rdrr.io/r/base/system.file.html">system.file</a></span>(<span class="st">"extdata"</span>,
                      <span class="st">"dataset1.csv"</span>, <span class="kw">package</span> <span class="kw">=</span> <span class="st">"coca"</span>), <span class="kw">row.names</span> <span class="kw">=</span> <span class="fl">1</span>))
<span class="no">data</span><span class="kw">[[</span><span class="fl">2</span>]] <span class="kw">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/matrix.html">as.matrix</a></span>(<span class="fu"><a href="https://rdrr.io/r/utils/read.table.html">read.csv</a></span>(<span class="fu"><a href="https://rdrr.io/r/base/system.file.html">system.file</a></span>(<span class="st">"extdata"</span>,
                      <span class="st">"dataset2.csv"</span>, <span class="kw">package</span> <span class="kw">=</span> <span class="st">"coca"</span>), <span class="kw">row.names</span> <span class="kw">=</span> <span class="fl">1</span>))
<span class="no">data</span><span class="kw">[[</span><span class="fl">3</span>]] <span class="kw">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/matrix.html">as.matrix</a></span>(<span class="fu"><a href="https://rdrr.io/r/utils/read.table.html">read.csv</a></span>(<span class="fu"><a href="https://rdrr.io/r/base/system.file.html">system.file</a></span>(<span class="st">"extdata"</span>,
                      <span class="st">"dataset3.csv"</span>, <span class="kw">package</span> <span class="kw">=</span> <span class="st">"coca"</span>), <span class="kw">row.names</span> <span class="kw">=</span> <span class="fl">1</span>))

<span class="co">### Build matrix of clusters</span>
<span class="no">outputBuildMOC</span> <span class="kw">&lt;-</span> <span class="kw pkg">coca</span><span class="kw ns">::</span><span class="fu"><a href="../reference/buildMOC.html">buildMOC</a></span>(<span class="no">data</span>, <span class="kw">M</span> <span class="kw">=</span> <span class="fl">3</span>, <span class="kw">K</span> <span class="kw">=</span> <span class="fl">5</span>, <span class="kw">distances</span> <span class="kw">=</span> <span class="st">"cor"</span>)

<span class="co">### Extract matrix of clusters and dataset indicator vector</span>
<span class="no">moc</span> <span class="kw">&lt;-</span> <span class="no">outputBuildMOC</span>$<span class="no">moc</span>
<span class="no">datasetIndicator</span> <span class="kw">&lt;-</span> <span class="no">outputBuildMOC</span>$<span class="no">datasetIndicator</span></pre></body></html></div>
<p>The package also contains a function that can be used to plot the resulting Matrix-Of-Clusters, <code>plotMOC</code>. Here we use as annotations the true cluster labels, but, in real applications, the <code>annotations</code> argument can take as input any dataframe with one element for each row of the matrix of clusters.</p>
<p>Please note that the row names of the dataframe must be the same must correspond to the observation names in each dataset, otherwise the annotation cells will be left empty by the plotting function. Moreover, each column of the dataframe that contains categorical variables must be defined with <code><a href="https://rdrr.io/r/base/factor.html">as.factor()</a></code> if you want each category to have a different colour (otherwise they will be treated as continuous variables and each category will have a different shade of the same colour).</p>
<div class="sourceCode" id="cb2"><html><body><pre class="r"><span class="co">### Prepare annotations</span>
<span class="no">true_labels</span> <span class="kw">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/matrix.html">as.matrix</a></span>(<span class="fu"><a href="https://rdrr.io/r/utils/read.table.html">read.csv</a></span>(<span class="fu"><a href="https://rdrr.io/r/base/system.file.html">system.file</a></span>(<span class="st">"extdata"</span>, <span class="st">"cluster_labels.csv"</span>,
                <span class="kw">package</span> <span class="kw">=</span> <span class="st">"coca"</span>), <span class="kw">row.names</span> <span class="kw">=</span> <span class="fl">1</span>))
<span class="no">annotations</span> <span class="kw">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/data.frame.html">data.frame</a></span>(<span class="kw">true_labels</span> <span class="kw">=</span> <span class="fu"><a href="https://rdrr.io/r/base/factor.html">as.factor</a></span>(<span class="no">true_labels</span>))

<span class="co">### Plot matrix of clusters</span>
<span class="kw pkg">coca</span><span class="kw ns">::</span><span class="fu"><a href="../reference/plotMOC.html">plotMOC</a></span>(<span class="no">moc</span>, <span class="no">datasetIndicator</span>, <span class="kw">annotations</span> <span class="kw">=</span> <span class="no">annotations</span>)</pre></body></html></div>
<p><img src="coca-vignette_files/figure-html/plot_moc-1.png" width="700"></p>
<p>Here the datasets don’t have names, so they have been assigned integer numbers. If available, you can specify cluster names; this will make the row names easier to interpret.</p>
<div class="sourceCode" id="cb3"><html><body><pre class="r"><span class="co">### Prepare annotations</span>
<span class="no">true_labels</span> <span class="kw">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/matrix.html">as.matrix</a></span>(<span class="fu"><a href="https://rdrr.io/r/utils/read.table.html">read.csv</a></span>(<span class="fu"><a href="https://rdrr.io/r/base/system.file.html">system.file</a></span>(<span class="st">"extdata"</span>, <span class="st">"cluster_labels.csv"</span>,
                <span class="kw">package</span> <span class="kw">=</span> <span class="st">"coca"</span>), <span class="kw">row.names</span> <span class="kw">=</span> <span class="fl">1</span>))
<span class="no">annotations</span> <span class="kw">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/data.frame.html">data.frame</a></span>(<span class="kw">true_labels</span> <span class="kw">=</span> <span class="fu"><a href="https://rdrr.io/r/base/factor.html">as.factor</a></span>(<span class="no">true_labels</span>))

<span class="co">### Set dataset names</span>
<span class="no">datasetNames</span> <span class="kw">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span>(<span class="fu"><a href="https://rdrr.io/r/base/rep.html">rep</a></span>(<span class="st">"A"</span>, <span class="fl">5</span>), <span class="fu"><a href="https://rdrr.io/r/base/rep.html">rep</a></span>(<span class="st">"B"</span>, <span class="fl">5</span>), <span class="fu"><a href="https://rdrr.io/r/base/rep.html">rep</a></span>(<span class="st">"C"</span>, <span class="fl">5</span>))

<span class="co">### Plot matrix of clusters</span>
<span class="kw pkg">coca</span><span class="kw ns">::</span><span class="fu"><a href="../reference/plotMOC.html">plotMOC</a></span>(<span class="no">moc</span>, <span class="no">datasetIndicator</span>, <span class="kw">datasetNames</span> <span class="kw">=</span> <span class="no">datasetNames</span>,
        <span class="kw">annotations</span> <span class="kw">=</span> <span class="no">annotations</span>)</pre></body></html></div>
<p><img src="coca-vignette_files/figure-html/plotmoc_wnames-1.png" width="700"></p>
<p>As you can see, the first part of each row name corresponds to the dataset, the second one to the cluster index. Moreover, each colour in the main matrix corresponds to one dataset (here dataset A is green, dataset B is yellow, and so on).</p>
</div>
<div id="cluster-of-clusters-analysis" class="section level1">
<h1 class="hasAnchor">
<a href="#cluster-of-clusters-analysis" class="anchor"></a>Cluster-Of-Clusters Analysis</h1>
<p>The MOC matrix is then used as input to consensus clustering (CC), an algorithm that was developed by Monti et al. (2003) to assess cluster stability when analysing a single dataset. The resulting consensus matrix is then used as the similarity matrix for a hierarchical clustering method (or any other distance-based clustering algorithm). These last two steps of the COCA algorithm are contained in the <code><a href="../reference/coca.html">coca()</a></code> function.</p>
<div class="sourceCode" id="cb4"><html><body><pre class="r"><span class="co">### COCA</span>

<span class="co"># Use COCA to find global clustering</span>
<span class="no">coca</span> <span class="kw">&lt;-</span> <span class="kw pkg">coca</span><span class="kw ns">::</span><span class="fu"><a href="../reference/coca.html">coca</a></span>(<span class="no">moc</span>, <span class="kw">K</span> <span class="kw">=</span> <span class="fl">5</span>)

<span class="co"># Compare clustering to the true labels</span>
<span class="no">ari</span> <span class="kw">&lt;-</span> <span class="kw pkg">mclust</span><span class="kw ns">::</span><span class="fu"><a href="https://mclust-org.github.io/mclust/reference/adjustedRandIndex.html">adjustedRandIndex</a></span>(<span class="no">true_labels</span>, <span class="no">coca</span>$<span class="no">clusterLabels</span>)
<span class="no">ari</span></pre></body></html></div>
<pre><code>## [1] 1</code></pre>
<div class="sourceCode" id="cb6"><html><body><pre class="r"><span class="co">### Plot the matrix of clusters with the newly found cluster labels</span>
<span class="no">annotations</span>$<span class="no">coca</span> <span class="kw">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/factor.html">as.factor</a></span>(<span class="no">coca</span>$<span class="no">clusterLabels</span>)
<span class="kw pkg">coca</span><span class="kw ns">::</span><span class="fu"><a href="../reference/plotMOC.html">plotMOC</a></span>(<span class="no">moc</span>, <span class="no">datasetIndicator</span>, <span class="kw">datasetNames</span> <span class="kw">=</span> <span class="no">datasetNames</span>,
        <span class="kw">annotations</span> <span class="kw">=</span> <span class="no">annotations</span>)</pre></body></html></div>
<p><img src="coca-vignette_files/figure-html/coca-1.png" width="700"></p>
<p>Here, we have computed the Adjusted Rand Index (ARI) to check how similar the two partitions of the data are. The ARI goes from -1 to 1: values close to 1 indicate very high similarity between the two partitions, values close to 0 indicate that the observed amount of similarities between the two partitions is the expected similarity between two random partitions, while negative values indicate that the two partitions are less similar than what would be expected by chance.</p>
<p>Again, if the number of clusters is not know a priori, the user can delegate the choice of K to the <code><a href="../reference/coca.html">coca()</a></code> function. There are two methods available here: the silhouette, where the distance between data points <em>i</em> and <em>j</em> is defined as <em>1</em> minus the <em>(i,j)</em>-th kernel entry in the final kernel matrix, and the delta area under the curve, which is the method suggested by Monti et al. (2003). Please note that the properties of the latter have not been assessed yet.</p>
<div class="sourceCode" id="cb7"><html><body><pre class="r"><span class="co"># Use COCA to find global clustering and chooose the number of clusters</span>
<span class="no">coca</span> <span class="kw">&lt;-</span> <span class="kw pkg">coca</span><span class="kw ns">::</span><span class="fu"><a href="../reference/coca.html">coca</a></span>(<span class="no">moc</span>, <span class="kw">maxK</span> <span class="kw">=</span> <span class="fl">10</span>, <span class="kw">hclustMethod</span> <span class="kw">=</span> <span class="st">"average"</span>)

<span class="co"># Compare clustering to the true labels</span>
<span class="no">ari</span> <span class="kw">&lt;-</span> <span class="kw pkg">mclust</span><span class="kw ns">::</span><span class="fu"><a href="https://mclust-org.github.io/mclust/reference/adjustedRandIndex.html">adjustedRandIndex</a></span>(<span class="no">true_labels</span>, <span class="no">coca</span>$<span class="no">clusterLabels</span>)
<span class="no">ari</span></pre></body></html></div>
<pre><code>## [1] 0.8117604</code></pre>
<div class="sourceCode" id="cb9"><html><body><pre class="r"><span class="co">### Plot the matrix of clusters with the newly found cluster labels</span>
<span class="no">annotations</span>$<span class="no">coca</span> <span class="kw">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/factor.html">as.factor</a></span>(<span class="no">coca</span>$<span class="no">clusterLabels</span>)
<span class="kw pkg">coca</span><span class="kw ns">::</span><span class="fu"><a href="../reference/plotMOC.html">plotMOC</a></span>(<span class="no">moc</span>, <span class="no">datasetIndicator</span>, <span class="kw">datasetNames</span> <span class="kw">=</span> <span class="no">datasetNames</span>,
        <span class="kw">annotations</span> <span class="kw">=</span> <span class="no">annotations</span>)</pre></body></html></div>
<p><img src="coca-vignette_files/figure-html/coca_unknownK-1.png" width="700"></p>
</div>
<div id="clustering-algorithms" class="section level1">
<h1 class="hasAnchor">
<a href="#clustering-algorithms" class="anchor"></a>Clustering algorithms</h1>
<p><strong>Inside <code><a href="../reference/buildMOC.html">buildMOC()</a></code></strong></p>
<p>In the first step, where a different clustering is found in each dataset, inside the <code><a href="../reference/buildMOC.html">buildMOC()</a></code> function, the clustering algorithms available are:</p>
<ul>
<li>k-means clustering (<code>method="kmeans"</code>), for which the <code><a href="https://rdrr.io/r/stats/kmeans.html">kmeans()</a></code> function of the <code>stats</code> R package is used (see <a href="https://www.rdocumentation.org/packages/stats/versions/3.6.2/topics/kmeans">this link</a> for the documentation);</li>
<li>hierarchical clustering (<code>method="hclust"</code>), for which the <code><a href="https://rdrr.io/r/stats/hclust.html">hclust()</a></code> function of the <code>stats</code> R package is used (see <a href="https://www.rdocumentation.org/packages/stats/versions/3.6.2/topics/hclust">this link</a> for the documentation);</li>
<li>partitioning around medoids (<code>method="pam"</code>), for which the <code>pam()</code> function of the <code>cluster</code> R package is used (see <a href="https://www.rdocumentation.org/packages/cluster/versions/2.1.0/topics/pam">this link</a> for the documentation).</li>
</ul>
<p>If you would like to use a different clustering algorithm for each of your datasets, you can pass a vector of strings to the <code>method</code> argument. For example, by setting <code>method = c("kmeans", "hclust", "pam")</code> the clusters in the first dataset will be determined using k-means clustering, in the second one hierarchical clustering, and so on.</p>
<p><strong>Inside <code><a href="../reference/coca.html">coca()</a></code></strong></p>
<p>To build the consensus clustering matrix, the default clustering algorithm is k-means, as implemented in the <code>stats</code> package. The default maximum number of iterations is 1000. If you would like to increase it, you can use the <code>maxIterKM</code> argument of the <code><a href="../reference/coca.html">coca()</a></code> function. Alternatively, one can select to use hierarchical clustering; this can be achieved by setting <code>ccClMethod = "hclust"</code>. The distance used with hierarchical clustering can be selected via the <code>ccDistHC</code> argument of the <code><a href="../reference/coca.html">coca()</a></code> function.</p>
<p>To find the final clustering based on the consensus matrix, hierarchical clustering is used. The agglomeration method used in this step can be chosen via the <code>hclustMethod</code> of the <code><a href="../reference/coca.html">coca()</a></code> function. The available options are the same as in the <code><a href="https://rdrr.io/r/stats/hclust.html">hclust()</a></code> function of the <code>stats</code> R package.</p>
</div>
<div id="warning-and-error-messages" class="section level1">
<h1 class="hasAnchor">
<a href="#warning-and-error-messages" class="anchor"></a>Warning and error messages</h1>
<p>If you get one or more error messages saying <code>did not converge in 1000 iterations</code> when using the <code><a href="../reference/coca.html">coca()</a></code> function, it means that you are using the k-means clustering algorithm and that it did not converge within the maximum number of iterations (the default is 1000). If you would like to change the maximum number of iterations or use a different clustering algorithm, please see the previous section.</p>
</div>
<div id="references" class="section level1">
<h1 class="hasAnchor">
<a href="#references" class="anchor"></a>References</h1>
<p>Aure, M.R., Vitelli, V., Jernström, S., Kumar, S., Krohn, M., Due, E.U., Haukaas, T.H., Leivonen, S.K., Vollan, H.K.M., Lüders, T. and Rødland, E. (2017). Integrative clustering reveals a novel split in the luminal A subtype of breast cancer with impact on outcome. Breast Cancer Research, 19(1), p.44.</p>
<p>Cabassi, A. and Kirk, P. D. W. (2020). Multiple kernel learning for integrative consensus clustering of ’omic datasets. Bioinformatics, btaa593.</p>
<p>Hoadley, K.A., Yau, C., Wolf, D.M., Cherniack, A.D., Tamborero, D., Ng, S., Leiserson, M.D., Niu, B., McLellan, M.D., Uzunangelov, V. and Zhang, J., 2014. Multiplatform analysis of 12 cancer types reveals molecular classification within and across tissues of origin. Cell, 158(4), pp.929-944.</p>
<p>Monti, S. et al. (2003). Consensus Clustering: A Resampling-Based Method for Class Discovery and Visualization of Gene. Machine Learning, 52(i), 91–118.</p>
<p>The Cancer Genome Atlas (2012). Comprehensive molecular portraits of human breast tumours. Nature, 487(7407), 61–70.</p>
</div>
  </div>

  <div class="col-md-3 hidden-xs hidden-sm" id="pkgdown-sidebar">

        <nav id="toc" data-toggle="toc"><h2 data-toc-skip>Contents</h2>
    </nav>
</div>

</div>



      <footer><div class="copyright">
  <p>Developed by <a href="https://alessandracabassi.wordpress.com">Alessandra Cabassi</a>.</p>
</div>

<div class="pkgdown">
  <p>Site built with <a href="https://pkgdown.r-lib.org/">pkgdown</a> 1.5.1.</p>
</div>

      </footer>
</div>

  


  </body>
</html>
